src/repl/scala/tools/nsc/interpreter/ReplGlobal.scala

```diff
@@ -40,6 +42,35 @@ trait ReplGlobal extends Global {
     }
   }
 
+  override protected def computeInternalPhases(): Unit = {
+    super.computeInternalPhases()
+    addToPhasesSet(wrapperCleanup, "Remove unused values from import wrappers to avoid unwanted capture of non-serializable objects")
+  }
+
+  private object wrapperCleanup extends Transform {
+    override val global: self.type = self
+    override val phaseName: String = "wrapper-cleanup"
+    /** Names of phases that must run before this phase. */
+    override val runsAfter: List[String] = List("refchecks")
```

Comment:
Should we also define `runsBefore` to avoid surprises? (https://github.com/scala/scala/pull/8426)